test/unit/test_buf.c

```diff
@@ -1055,6 +1055,26 @@ bool test_rz_buf_get_string_nothing(void) {
 	mu_end;
 }
 
+bool test_rz_buf_get_nstring(void) {
```

Comment:
add also a call to `rz_buf_get_nstring` with a size >= 32 but less than 64. You should check that the returned string is not longer than the expected size.

Reply:
I add a test, what do you think?

Reply:
I think you should actually test that `strlen(s) < x`, where `x` is the size you passed to `rz_buf_get_nstring`

Reply:
* test with string not found
* test with string found (strlen(s) == x)
* test with the full buffer
* test with string found (strlen(s) < x and strlen(s) == 32)
* test with string found (strlen(s) == x and strlen(s) == 32)

Reply:
mmm you didn't check that strlen(s) < x.

You could do:
```c
mu_assert_true(strlen(s) < x, "xxxx");
```

Reply:
mu_assert_streq_free should assert this right?

Reply:
In theory it checks that it's the same string pointed by the `expected` argument... I'm not 100% convinced this is the same as making sure that the returned string is of the maximum expected length.

Reply:
I add `mu_assert_true`

Reply:
I guess it is alright! Thanks!